File: php/classes/Empresa.php
<?php 

require_once __DIR__ . "/enum/EmpresaStatus.php";
require_once __DIR__ . "/Usuario.php";

class Empresa Extends Usuario
{
    private string $cnpj;
    private EmpresaStatus $status;

    public function __construct(int $id, string $nome, string $cnpj, string $email, EmpresaStatus $status = EmpresaStatus::PENDENTE)
    {
        $this->setId($id);
        $this->nome = $nome;
        $this->cnpj = $cnpj;
        $this->email = $email;
        $this->status = $status;
    }
    

    public function getNome(): string
    {
        return $this->nome;
    }

    public function setNome(string $nome): void
    {
        $this->nome = $nome;
    }

    public function getCnpj(): string
    {
        return $this->cnpj;
    }

    public function setCnpj(string $cnpj): void
    {
        $this->cnpj = $cnpj;
    }

    public function getEmail(): string
    {
        return $this->email;
    }

    public function setEmail(string $email): void
    {
        $this->email = $email;
    }

    public function getSenha(): string
    {
        return $this->senha;
    }

    public function setSenha(string $senha): void
    {
        $this->senha = $senha;
    }

    public function getStatus(): EmpresaStatus
    {
        return $this->status;
    }

    public function setStatus(EmpresaStatus $status): void
    {
        $this->status = $status;
    }

}

File: java/src/main/java/model/Empresa.java
package model;

public class Empresa extends EntidadeBase {

    private String nome;
    private String cnpj;
    private String email;
    private String senhaHash;
    private String status;

    public Empresa() { super(); }

    public Empresa(Integer id, String nome, String cnpj, String email, String status) {
        super(id);
        this.nome   = nome;
        this.cnpj   = cnpj;
        this.email  = email;
        this.status = status;
    }

    public String getNome()   { return nome; }
    public void setNome(String nome) { this.nome = nome; }

    public String getCnpj()   { return cnpj; }
    public void setCnpj(String cnpj) { this.cnpj = cnpj; }

    public String getEmail()  { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getSenhaHash() { return senhaHash; }
    public void setSenhaHash(String senhaHash) { this.senhaHash = senhaHash; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    @Override
    public String toString() { return nome + " (" + status + ")"; }
}